bgpd/bgp_packet.c

```diff
@@ -3044,6 +3083,83 @@ static void bgp_dynamic_capability_addpath(uint8_t *pnt, int action,
 	}
 }
 
+static void bgp_dynamic_capability_fqdn(uint8_t *pnt, int action,
+					struct capability_header *hdr,
+					struct peer *peer)
+{
+	uint8_t *data = pnt + 3;
+	uint8_t *end = data + hdr->length;
+	char str[BGP_MAX_HOSTNAME + 1] = {};
+	uint8_t len;
+
+	if (action == CAPABILITY_ACTION_SET) {
+		/* hostname */
+		len = *data;
```

Comment:
Don't we need a data + 1 check here for the reading of length?

Reply:
Added this check, thanks.